cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerServiceTest.java

```diff
@@ -165,6 +172,70 @@ public void testSimpleSchedulerLifecycle() throws Exception {
     checkState(Scheduler.ScheduleState.NOT_FOUND, scheduleIds);
   }
 
+  @Test
+  public void testScheduleUpdate() throws Exception {
+    // test time schedule update
+    testScheduleUpdate(TIME_SCHEDULE_1, UPDATED_TIME_SCHEDULE_1, TimeUnit.MINUTES.toMillis(10));
+    // test stream size schedule update
+    testScheduleUpdate(DATA_SCHEDULE_2, UPDATED_DATA_SCHEDULE_2, null);
+  }
+
+  private void testScheduleUpdate(Schedule oldSchedule, Schedule newSchedule, @Nullable Long durationInMillis)
```

Comment:
why is the last param nullable? It seems that its passed to `verifyUpdatedNextRuntime` below on line 225, on which it is not nullable.